assimilation_code/modules/utilities/netcdf_utilities_mod.f90

```diff
@@ -1168,25 +1263,24 @@ end subroutine nc_put_int_1d
 
 !--------------------------------------------------------------------
 
-subroutine nc_put_single_real_1d(ncid, varname, varindex, varval, context, filename)
+subroutine nc_put_real_0d(ncid, varname, varval, context, filename)
```

Comment:
hi tim - have you tried compiling all the converters with this version of the netcdf utilities?  i believe the "put single 1d" routines were not intended to write scalars, but to overwrite a single value in a 1d array.  thus varindex is needed.  if the overloading doesn't break, perhaps you need to keep both? 

your question about using names with real vs float is a valid one.  i was inconsistent in terminology.  float and double are C conventions, and match how netcdf names them i think; real is fortran but doesn't indicate size.  if you want to rename them all to float/double i think that's transparent to the calling code.

Reply:
I have run this through the entire test suite, which did check this code.  There is still a nc_put_real_1d() routine, which could be used to stuff an array of length one (technically not a scalar) into some slot in a netCDF array variable. Stuffing a scalar into  a slot in an array is - I think - unsupported as is, but the only place that was happening was in the lorenz_96_2scale model - which now writes an array into an array. None of the observation converters had anything to say about it. I suspect 99% of them are still using the netCDF routines in obs_converters/utilities/obs_utilities_mod.f90

I'll tackle the renaming at a future time, this PR is/was spiraling out of control and I'm already at the limit of the scope I wanted.

Reply:
ok.  i see the original usage in lorenz_96_2scale.  you changed it to put the whole arrays in a single call?  sounds good.  (i couldn't imagine that i'd invent this variation unless it was used someplace.)